### DISCUSSION TITLE
failed to create symbolic link

### DISCUSSION THREAD

**Discussion by k-msalehi** [deployphp/deployer#3727] [Help needed] [answered]
I'm using windows 10, deployer version v7.3.3 and git bash.
When I run `vendor/bin/dep deploy`  I get `failed to create symbolic link` error,
full output:
```
$vendor/bin/dep deploy
task deploy:info
[localhost] info deploying HEAD
task deploy:setup
task deploy:lock
task deploy:release
[localhost]  error  in release.php on line 131:
[localhost] run cd C:/laragon/www/dep-test && (ln -nfs --relative releases/9 C:/laragon/www/dep-test/release)
[localhost] err ln: failed to create symbolic link 'C:/laragon/www/dep-test/release/9': No such file or directory
[localhost] exit code 1 (General error)
ERROR: Task deploy:release failed!
task deploy:failed
task deploy:unlock
```

My code
```php
namespace Deployer;

require 'recipe/composer.php';

// Config
set('keep_releases', 3);
set('repository', 'https://github.com/msalehi-d/PDate.git');

// Hosts
localhost()->set('deploy_path', 'C:/laragon/www/dep-test');

after('deploy:failed', 'deploy:unlock');
```

I've tried to create directory before deploy but I get this warning: ` Release name "8" already exists.`

**Comment by acharseth** [comment]
Windows does not support creating symbolic links out of the box. Even if it possible to enable it, it is a security issue.
You could set up WSL in Windows running Ubuntu, or some other Linux, and running deployer there. Quite a bit of work to setup however.

**Comment by k-msalehi** [comment]
@acharseth thnaks, So in Windows there is no way except WSL to use deployer?

**Reply by acharseth** [reply, marked as answer]
Well, if you are an administrator and willing to take the risk with the security issues then you could create symbolic links in Windows. See https://learn.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/create-symbolic-links for more information.

Otherwise I guess you must come up with another way to get zero down time and/or create complete copies of deployed code and configurations by changing the way Deployer works. May be @antonmedv have other solutions?
